### ISSUE TITLE
`no-fn-reference-in-iterator` false-positive

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/eslint-plugin-unicorn#694]
It triggered on:

https://github.com/sindresorhus/got/blob/731e4fa235ae9b43e03e74fee34b8b32d8ebb705/benchmark/index.ts#L165

```
benchmark/index.ts:165:40
  ✖   165:40  Do not pass function directly to .filter(…).                                     unicorn/no-fn-reference-in-iterator
  ✖   165:55  Do not pass function directly to .map(…).                                        unicorn/no-fn-reference-in-iterator
```

**Comment by fisker** [comment]
What is that `this`? `Benchmark.Suite`? We can exclude your case, by checking the argument type is not `Literal`

**Comment by sindresorhus** [comment]
Couldn't we also filter out use of `this.`?

**Comment by sindresorhus** [comment]
> What is that this? Benchmark.Suite?

I think so, but doesn't really matter. We cannot know what `this` is with ESLint anyway.

**Comment by fisker** [comment]
This could be `Array`.

**Comment by sindresorhus** [comment]
It could, but very unlikely. I think we should optimize for the common-case where it's not.

**Comment by fisker** [comment]
I'm not sure about `this.`.

I'm sending #697 to ignore cases obviously not a function reference.

**Comment by fisker** [comment]
And to ignore `this.` need more work, `this.filter().map()` if we only check callee is not `this`, `.filter()` will ignored, but `.map()` won't because `callee` of `.map()` is a `MemberExpression`, maybe I'll fix it tomorrow.

**Comment by fisker** [comment]
On second thought, we can't filter the `.map` call in  `this.filter().map()`, because `this.filter()` can return an array, and we can't only check the root of the chain call.
In this case `this.foo().map()`, `this.foo()` can be an array.

**Comment by fisker** [comment]
Should be fixed by #697 and #699
